probe-rs/src/architecture/arm/core/armv8a.rs

```diff
@@ -609,6 +609,12 @@ impl<'probe> Armv8a<'probe> {
     fn read_cpu_memory_aarch32_32(&mut self, address: u64) -> Result<u32, Error> {
         let address = valid_32bit_address(address)?;
 
+        // save halt status
+        let original_halt_status = self.core_halted()?;
```

Comment:
I feel like if you need to copy this snippet 6 times, you should instead add a `fn with_core_halted<R>(impl FnOnce(&mut Self) -> R) -> R` wrapper instead. Less for us to review, less opportunities to forget resuming :)

Reply:
Thank you for your review :)
I added a wrapper function https://github.com/probe-rs/probe-rs/pull/2269/commits/af02d87b1de1be22eea509b74b75736be86832e2
Could you please review it again?

If it is OK, can I rebase these commits into one?